From ad60b4e56b7f9803d076284715530dfe47945683 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Wed, 27 May 2020 15:38:36 +0530
Subject: [PATCH 3/3] Avoid-kernel-hang

This patch will override CPU_SLEEP, busfreq and resmem devicetree nodes to
avoid kernel hang issue.

Since GPC is not used now, "arm,psci-suspend-param" flag is set to zero in
CPU_SLEEP node.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 .../boot/dts/adlink/adlink-lec-imx8m.dts      | 44 +++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
index 8c3fdec62cd1..dffe07d0278e 100755
--- a/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8m.dts
@@ -13,6 +13,8 @@
 	model = "ADLINK LEC i.MX8M";
 	compatible = "adlink,lec-imx8m", "fsl,imx8mq";
 
+	interrupt-parent = <&gic>;
+
 	chosen {
 		stdout-path = &uart1;
 	};
@@ -192,6 +194,48 @@
 	};
 };
 
+&CPU_SLEEP {
+	/* We are not using GPC for now, need set 0 to avoid hang */
+	arm,psci-suspend-param = <0x0>;
+};
+
+&{/busfreq} {
+	/* Disable busfreq, to avoid 1st Linux busfreq crash other inmates */
+	status = "disabled";
+};
+
+&resmem {
+	jh_reserved: jh@fdc00000 {
+		no-map;
+		reg = <0 0xfdc00000 0x0 0x400000>;
+	};
+
+	inmate_reserved: inmate@c0000000 {
+		no-map;
+		reg = <0 0xc0000000 0x0 0x3dc00000>;
+	};
+
+	loader_reserved: loader@bff00000 {
+		no-map;
+		reg = <0 0xbff00000 0x0 0x00100000>;
+	};
+
+	ivshmem_reserved: ivshmem@bfe00000 {
+		no-map;
+		reg = <0 0xbfe00000 0x0 0x00100000>;
+	};
+
+	ivshmem2_reserved: ivshmem2@bfd00000 {
+		no-map;
+		reg = <0 0xbfd00000 0x0 0x00100000>;
+	};
+
+	pci_reserved: pci@bfc00000 {
+		no-map;
+		reg = <0 0xbfb00000 0x0 0x00200000>;
+	};
+};
+
 &A53_0 {
 	cpu-supply = <&buck2_reg>;
 };
-- 
2.17.1

